File: DS/DS-Session1/Stack/Stack.java

public interface Stack {

	/**
	 * Push Data inside Stack
	 * @param data - Date value
	 */
	public void push(Data data);
	
	/**
	 * Pop top element of the stack
	 * @param data - data value
	 * @return -  true if element popped otherwise false
	 */
	public void pop();
	
	/**
	 * 
	 * @return - Top element of the stack
	 */
	public Data getPeak();
	
	/**
	 * 
	 * @return - size of the Stack
	 */
	public int getSize();
	
	/**
	 * @return - True if stack is empty otherwise false;
	 */
	public boolean isEmpty();
}

File: UI/UI-Session-8/stack.js
//Stack class
function Stack(){
    this.arr = [];
}

/**
 * insert data at the end of the stack
 * @param {data} data 
 */
Stack.prototype.push = function(data){
    this.arr.push(data);
};

/**
 * remove top element of the stack
 */
Stack.prototype.pop = function(){
    return this.arr.pop();
};

/**
 * return size of the stack
 */
Stack.prototype.size = function(){
    return this.arr.length;
};

/**
 * return top element of the stack
 */
Stack.prototype.peek = function(){
    return this.arr[this.arr.length-1];
};

function main(){
    var stack = new Stack();
    stack.push("Ramjilal");
    stack.push("Vaibhav");
    console.log(stack.peek());
    console.log(stack.size());
    console.log(stack.pop());
    console.log(stack.peek());
}

main();